src/test/java/teammates/test/cases/webapi/PostStudentProfilePictureActionTest.java

```diff
@@ -0,0 +1,79 @@
+package teammates.test.cases.webapi;
+
+import org.apache.http.HttpStatus;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.AccountAttributes;
+import teammates.common.exception.InvalidHttpRequestBodyException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.action.JsonResult;
+import teammates.ui.webapi.action.PostStudentProfilePictureAction;
+import teammates.ui.webapi.output.StudentProfilePictureResults;
+
+/**
+ * SUT: {@link PostStudentProfilePictureAction}.
+ */
+public class PostStudentProfilePictureActionTest extends BaseActionTest<PostStudentProfilePictureAction> {
+    @Override
+    protected String getActionUri() {
+        return Const.ResourceURIs.STUDENT_PROFILE_PICTURE;
+    }
+
+    @Override
+    protected String getRequestMethod() {
+        return POST;
+    }
+
+    @Override
+    @Test
+    public void testExecute() throws Exception {
+        AccountAttributes student1 = typicalBundle.accounts.get("student1InCourse1");
+        loginAsStudent(student1.googleId);
+
+        ______TS("Typical case: upload profile picture operation successful");
+
+        String filePath = "src/test/resources/images/profile_pic.png";
+        PostStudentProfilePictureAction action = getActionWithParts("studentprofilephoto", filePath);
+        JsonResult result = getJsonResult(action);
+
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+
+        StudentProfilePictureResults output = (StudentProfilePictureResults) result.getOutput();
+        String resultPictureKey = output.getPictureKey();
+        String newPictureKey = logic.getStudentProfile(student1.googleId).pictureKey;
+
+        assertNotNull(resultPictureKey);
+        assertNotEquals(resultPictureKey, "");
+        assertEquals(resultPictureKey, newPictureKey);
+
+        ______TS("Typical case: profile picture is null");
+
+        PostStudentProfilePictureAction nullProfilePicAction =
+                getActionWithParts("studentprofilephoto", "");
+
+        assertThrows(NullPointerException.class, () -> nullProfilePicAction.execute());
```

Comment:
`NullPointerException` getting thrown does not sound right. What's actually happening here?

If you're trying to test a null profile picture part, you should instead use the normal `getAction`.

Reply:
The action will hit the first line and throw a NPE, ignoring the rest of the lines:
```
            Part image = req.getPart("studentprofilephoto");
            if (image == null) {
                throw new InvalidHttpRequestBodyException(Const.StatusMessages.STUDENT_PROFILE_NO_PICTURE_GIVEN);
            }
```

But yup, should use a normal action to simulate null profile pic part.